pkg/controller/gitopsservice/dependency/dependency.go

```diff
@@ -0,0 +1,143 @@
+package dependency
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	cicdNs                 = "cicd"
+	sealedSecretsSubName   = "sealed-secrets-operator-helm"
+	argocdSubName          = "argocd-operator"
+	sealedSecretsGroupName = "sealed-secrets-operator-group"
+	argocdGroupName        = "argocd-operator-group"
+)
+
+var log = logf.Log.WithName("gitops_dependencies")
+
+// Dependency represents an instance of GitOps dependency
+type Dependency struct {
+	client client.Client
+	prefix string
+	log    logr.Logger
+}
+
+// NewClient create a new instance of GitOps dependencies
+func NewClient(client client.Client, prefix string) *Dependency {
+	return &Dependency{
+		client: client,
+		prefix: prefix,
+		log:    log.WithName("GitOps Dependencies"),
+	}
+}
+
+// Install the dependencies required by GitOps
+func (d *Dependency) Install() error {
+	d.log.Info("Installing GitOps dependencies")
+	ctx := context.Background()
+
+	operators := []operatorResource{newSealedSecretsOperator(d.prefix), newArgoCDOperator(d.prefix)}
+
+	// TODO: Install each operator using a separate goroutine to improve installation performance
+	for _, operator := range operators {
+		ns := operator.GetNamespace()
+		d.log.Info("Creating Namespace", "Namespace.Name", ns.Name)
+		err := d.createResourceIfAbsent(ctx, operator.GetNamespace(), types.NamespacedName{Name: ns.Name})
+		if err != nil {
+			return err
+		}
+
+		operatorGroup := operator.GetOperatorGroup()
+		d.log.Info("Creating OperatorGroup", "OperatorGroup.Name", operatorGroup.Name)
+		err = d.createResourceIfAbsent(ctx, operator.GetOperatorGroup(), types.NamespacedName{Name: operatorGroup.Name, Namespace: operatorGroup.Namespace})
+		if err != nil {
+			return err
+		}
+
+		subscription := operator.GetSubscription()
+		d.log.Info("Creating Subscription", "Subscription.Name", subscription.Name)
+		err = d.createResourceIfAbsent(ctx, operator.GetSubscription(), types.NamespacedName{Name: subscription.Name, Namespace: subscription.Namespace})
+		if err != nil {
+			return err
+		}
+
+		// TODO: Wait for the operator to install ie. check if operator CSV has success status(csv.spec.phase == Succeeded)
+
+		// TODO: Create the operator CR
+	}
+
+	return nil
+}
+
+func (d *Dependency) createResourceIfAbsent(ctx context.Context, obj runtime.Object, ns types.NamespacedName) error {
+	err := d.client.Get(ctx, ns, obj)
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
You could probably improve this by using an expression switch and cases for the options (and `default:` returning the error).